x/evm/watcher/watcher.go

```diff
@@ -173,8 +174,10 @@ func (w *Watcher) DeleteAccount(addr sdk.AccAddress) {
 	}
 	key1 := GetMsgAccountKey(addr.Bytes())
 	key2 := append(prefixRpcDb, key1...)
+	w.delAccountMtx.Lock()
```

Comment:
I think there has a risk that may decrease the performance of concurrent executing because this lock will turn parallel mode to serial mode